sklearn/metrics/_classification.py

```diff
@@ -3614,10 +3614,13 @@ def _validate_binary_probabilistic_prediction(y_true, y_prob, sample_weight, pos
     try:
         pos_label = _check_pos_label_consistency(pos_label, y_true)
     except ValueError:
-        classes = np.unique(y_true)
-        if classes.dtype.kind not in ("O", "U", "S"):
-            # for backward compatibility, if classes are not string then
-            # `pos_label` will correspond to the greater label
+        xp_y_true, _ = get_namespace(y_true)
+        classes = xp_y_true.unique_values(y_true)
+        if not (_is_numpy_namespace(xp_y_true) and classes.dtype.kind in "OUS"):
```

Comment:
Note this relies on lazy evaluation - if `_is_numpy_namespace(xp_y_true)` is False, the second condition is not evaluated.

Also note when namespace is not numpy, this evaluates to True.